packages/core/src/app/shipping/ShippingFormFooter.tsx

```diff
@@ -18,19 +24,50 @@ export interface ShippingFormFooterProps {
     isLoading: boolean;
 }
 
-class ShippingFormFooter extends PureComponent<ShippingFormFooterProps> {
+class ShippingFormFooter extends PureComponent<ShippingFormFooterProps & ExtensionContextProps> {
+    private _isRegionInUse = false;
+
+    componentWillUnmount(): void {
+        const { extensionService } = this.props;
+
+        if (this._isRegionInUse) {
+            extensionService.removeListeners(ExtensionRegion.ShippingShippingAddressFormAfter);
+        }
+    }
+
+    async componentDidMount(): Promise<void> {
+        const { extensionService } = this.props;
+
+        if (this._isRegionInUse) {
```

Comment:
Do you need this check or the need to keep track of `_isRegionInUse` in here? I think `ExtensionService#renderExtension` method already does a check so it won't render it again if the region is already occupied?

Reply:
`isRegionInUse` also represents the extension experiment flag.
So here is to skip the function if the experiment flag is `off`.

Reply:
I suggest moving the experiment check in `ExtensionService#renderExtension` as well

Reply:
Sure, can do that in a follow-up SDK PR.
Regarding `isRegionInUse`, do we want to avoid putting an empty div?
```
            {isRegionInUse && (
                <div id={ExtensionRegionContainer.ShippingShippingAddressFormBefore} />
            )}
```

Reply:
> do we want to avoid putting an empty div?

Yes we want to, but not by setting a property inside `render` method and reading from it.

Reply:
I've converted `ShippingFormFooter` into a functional component, so `isRegionInUse` is assigned differently now.
https://github.com/bigcommerce/checkout-js/blob/994a53343bd3dc4af0f2407e7cf33076088c88f2/packages/core/src/app/shipping/ShippingFormFooter.tsx#L32